### ISSUE TITLE
[PT2.2] PyTorch dispacthing non core ATen ops in .compile mode for torch ops.

### ISSUE THREAD

**Issue by rbhandarihabana** [pytorch/pytorch#121967]
### 🐛 Describe the bug

PyTorch dispacthing non core ATen ops in .compile mode for torch ops. 
Same non core ATen ops are observed as part of our backend.

**List of non core ATen ops observed are:**
```
addmv.default
addbmm.default
cumprod.default
median.default
min.default
```

### Error logs

aten ops getting involved for addmv ['addmv.default']
aten ops getting involved for addbmm ['addbmm.default']
aten ops getting involved for cumprod ['cumprod.default']
aten ops getting involved for median ['median.default']
aten ops getting involved for min ['min.default']

### Minified repro

```
import torch
from torch._decomp import core_aten_decompositions
from torch._dynamo.backends.common import aot_autograd

fx_info = {"fx_aten_ops": []}

def inner_compiler(fx_module: torch.fx.GraphModule, inputs):
    for node in fx_module.graph.nodes:
        if node.op == "call_function":
            fx_info["fx_aten_ops"].append(node.target.__name__)
    return fx_module

# UT for addmv ATen op
def addmv_model(M, mat, vec):
  return torch.addmv(M, mat, vec)

M = torch.randn(2)
mat = torch.randn(2, 3)
vec = torch.randn(3)

fx_trace_backend = aot_autograd(
    fw_compiler=inner_compiler, decompositions=core_aten_decompositions()
)
model_compiled = torch.compile(addmv_model, backend=fx_trace_backend)
result_fwd = model_compiled(M, mat, vec)

print(f"aten ops getting involved for addmv {fx_info['fx_aten_ops']}")

# UT for addbmm ATen op
def addbmm_model(M, batch1, batch2):
  return torch.addbmm(M, batch1, batch2)

M = torch.randn(3, 5)
batch1 = torch.randn(10, 3, 4)
batch2 = torch.randn(10, 4, 5)

fx_info = {"fx_aten_ops": []}
fx_trace_backend = aot_autograd(
    fw_compiler=inner_compiler, decompositions=core_aten_decompositions()
)
model_compiled = torch.compile(addbmm_model, backend=fx_trace_backend)
result_fwd = model_compiled(M, batch1, batch2)
print(f"aten ops getting involved for addbmm {fx_info['fx_aten_ops']}")

# UT for cumprod ATen op
def cumprod_model(ifm):
  return torch.cumprod(ifm, dim=0)

ifm = torch.randn(10)

fx_info = {"fx_aten_ops": []}
fx_trace_backend = aot_autograd(
    fw_compiler=inner_compiler, decompositions=core_aten_decompositions()
)
model_compiled = torch.compile(cumprod_model, backend=fx_trace_backend)
result_fwd = model_compiled(ifm)
print(f"aten ops getting involved for cumprod {fx_info['fx_aten_ops']}")


# UT for median ATen op
def median_model(ifm):
  return torch.median(ifm)

ifm = torch.randn(10)

fx_info = {"fx_aten_ops": []}
fx_trace_backend = aot_autograd(
    fw_compiler=inner_compiler, decompositions=core_aten_decompositions()
)
model_compiled = torch.compile(median_model, backend=fx_trace_backend)
result_fwd = model_compiled(ifm)
print(f"aten ops getting involved for median {fx_info['fx_aten_ops']}")


# UT for min ATen op
def min_model(ifm):
  return torch.min(ifm)

ifm = torch.randn(10)

fx_info = {"fx_aten_ops": []}
fx_trace_backend = aot_autograd(
    fw_compiler=inner_compiler, decompositions=core_aten_decompositions()
)
model_compiled = torch.compile(min_model, backend=fx_trace_backend)
result_fwd = model_compiled(ifm)
print(f"aten ops getting involved for min {fx_info['fx_aten_ops']}")

```

### Versions

[pip3] numpy==1.26.4
[pip3] torch==2.2.0a0
[pip3] torchaudio==2.2.0
[pip3] torchdata==0.7.1
[pip3] torchtext==0.17.0
[pip3] torchvision==0.17.0

**Comment by anijain2305** [comment]
@rbhandarihabana The ops are still core aten ops

If you do TORCH_LOGS="aot", you will see the graphs.

You will see ops like `torch.ops.aten.addmv.default`


~~~
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs] TRACED GRAPH
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]  ===== Forward graph 0 =====
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]  /data/users/anijain/pytorch/torch/fx/_lazy_graph_module.py class <lambda>(torch.nn.Module):
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]     def forward(self, arg0_1: "f32[2]", arg1_1: "f32[2, 3]", arg2_1: "f32[3]"):
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]         # File: /data/users/anijain/pytorch/examples/67.py:15 in addmv_model, code: return torch.addmv(M, mat, vec)
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]         addmv: "f32[2]" = torch.ops.aten.addmv.default(arg0_1, arg1_1, arg2_1);  arg0_1 = arg1_1 = arg2_1 = None
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]         return (addmv,)
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]
I0318 10:03:00.938000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [0/0] [__aot_graphs]
aten ops getting involved for addmv ['addmv.default']
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs] TRACED GRAPH
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]  ===== Forward graph 1 =====
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]  /data/users/anijain/pytorch/torch/fx/_lazy_graph_module.py class <lambda>(torch.nn.Module):
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]     def forward(self, arg0_1: "f32[3, 5]", arg1_1: "f32[10, 3, 4]", arg2_1: "f32[10, 4, 5]"):
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]         # File: /data/users/anijain/pytorch/examples/67.py:32 in addbmm_model, code: return torch.addbmm(M, batch1, batch2)
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]         addbmm: "f32[3, 5]" = torch.ops.aten.addbmm.default(arg0_1, arg1_1, arg2_1);  arg0_1 = arg1_1 = arg2_1 = None
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]         return (addbmm,)
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]
I0318 10:03:00.973000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [1/0] [__aot_graphs]
aten ops getting involved for addbmm ['addbmm.default']
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs] TRACED GRAPH
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]  ===== Forward graph 2 =====
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]  /data/users/anijain/pytorch/torch/fx/_lazy_graph_module.py class <lambda>(torch.nn.Module):
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]     def forward(self, arg0_1: "f32[10]"):
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]         # File: /data/users/anijain/pytorch/examples/67.py:48 in cumprod_model, code: return torch.cumprod(ifm, dim=0)
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]         cumprod: "f32[10]" = torch.ops.aten.cumprod.default(arg0_1, 0);  arg0_1 = None
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]         return (cumprod,)
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]
I0318 10:03:00.999000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [2/0] [__aot_graphs]
aten ops getting involved for cumprod ['cumprod.default']
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs] TRACED GRAPH
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]  ===== Forward graph 3 =====
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]  /data/users/anijain/pytorch/torch/fx/_lazy_graph_module.py class <lambda>(torch.nn.Module):
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]     def forward(self, arg0_1: "f32[10]"):
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]         # File: /data/users/anijain/pytorch/examples/67.py:63 in median_model, code: return torch.median(ifm)
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]         median: "f32[]" = torch.ops.aten.median.default(arg0_1);  arg0_1 = None
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]         return (median,)
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]
I0318 10:03:01.022000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [3/0] [__aot_graphs]
aten ops getting involved for median ['median.default']
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs] TRACED GRAPH
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]  ===== Forward graph 4 =====
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]  /data/users/anijain/pytorch/torch/fx/_lazy_graph_module.py class <lambda>(torch.nn.Module):
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]     def forward(self, arg0_1: "f32[10]"):
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]         # File: /data/users/anijain/pytorch/examples/67.py:78 in min_model, code: return torch.min(ifm)
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]         min_1: "f32[]" = torch.ops.aten.min.default(arg0_1);  arg0_1 = None
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]         return (min_1,)
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]
I0318 10:03:01.043000 140373459481728 torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:109] [4/0] [__aot_graphs]
aten ops getting involved for min ['min.default']
~~~

**Comment by rbhandarihabana** [comment]
@anijain2305  We are refereeing to the page https://pytorch.org/docs/main/torch.compiler_ir.html for core ATen ops and these are not part of this document.

Is there any place as part of PyTorch these ops are mentioned as core ATen ops?

**Comment by rbabukv** [comment]
@SujoysGithub @jgong5

**Comment by jthakurH** [comment]
Hi @anijain2305 ,

Can you share the the list to look for core Aten ops?

**Comment by jthakurH** [comment]
@anijain2305  any update on this?

**Comment by sujoysaraswati** [comment]
As per the [NOTE [Core ATen Ops]](https://github.com/pytorch/pytorch/blob/aaef246c74b964ba43f051a4a3e484d25d418f44/aten/src/ATen/native/tags.yaml#L50) " -
"Core aten ops is a subset of aten ops that remains after aten-to-aten decomposition and
          functionalization pass. Core aten ops are fully functional and adhere to single static
          assignment (SSA): this implies there will be no `inplace` or `_out` variants in this opset.
          This opset is designed to serve as the functional IR to interface with compiler backends."

However, the [core_aten_decompositions](https://github.com/pytorch/pytorch/blob/b63f6f78dc8a4e6f059b9abc4eb1eed1d490bcfc/torch/_decomp/__init__.py#L253) doesn't get decompositions for many non-core ATen ops. The example here, addmv is not marked as a core ATen IR op in https://pytorch.org/docs/stable/torch.compiler_ir.html#core-aten-ir. It has a [decomposition registered](https://github.com/pytorch/pytorch/blob/aaef246c74b964ba43f051a4a3e484d25d418f44/torch/_decomp/decompositions.py#L1454) , but core_aten_decomposition doesn't select this op to get decomposed into the core ATen IR ops.

Does this mean that FX graph will not follow only core ATen op lists by default?

**Comment by v0i0** [comment]
@anijain2305 is this still current? seems like we should either add `addmv` etc to the core aten decomposition list, or update the documentation to list those operations as part of the aten  core set?
